financial-connections/src/main/java/com/stripe/android/financialconnections/analytics/FinancialConnectionsInternalEvent.kt

```diff
@@ -0,0 +1,416 @@
+            StartVerificationError("StartVerificationSessionError"),
+            MarkLinkVerifiedError("MarkLinkStepUpAuthenticationVerifiedError"),
+        }
+    }
+
+    class ClickDone(
+        pane: Pane,
+    ) : FinancialConnectionsInternalEvent(
+        name = "click.done",
+        mapOf(
+            "pane" to pane.value,
+        ).filterNotNullValues()
+    )
+
+    class Error(
+        pane: Pane,
+        exception: Throwable,
+        extraMessage: String? = null,
+    ) : FinancialConnectionsInternalEvent(
+        name = when (exception) {
+            is FinancialConnectionsError,
+            is WebAuthFlowFailedException,
+            is OTPError -> "error.expected"
+
+            else -> "error.unexpected"
+        },
+        params = (
+            mapOf("pane" to pane.value)
+                .plus(exception.toEventParams(extraMessage))
+                .filterNotNullValues()
+            )
+    )
+
+    class Exposure(
+        experimentName: String,
+        assignmentEventId: String,
+        accountHolderId: String
+    ) : FinancialConnectionsInternalEvent(
+        name = "preloaded_experiment_retrieved",
+        params = mapOf(
+            "experiment_retrieved" to experimentName,
+            "arb_id" to assignmentEventId,
+            "account_holder_id" to accountHolderId
+        ).filterNotNullValues(),
+        includePrefix = false,
+    )
+
+    class AuthSessionUrlReceived(url: String, status: String, authSessionId: String?) :
+        FinancialConnectionsInternalEvent(
+            name = "auth_session.url_received",
+            params = mapOf(
+                "status" to status,
+                "url" to url,
+                "auth_session_id" to (authSessionId ?: "")
+            ).filterNotNullValues(),
+        )
+
+    class AuthSessionRetrieved(nextPane: Pane, authSessionId: String) :
+        FinancialConnectionsInternalEvent(
+            name = "auth_session.retrieved",
+            params = mapOf(
+                "next_pane" to nextPane.value,
+                "auth_session_id" to authSessionId,
+            ).filterNotNullValues(),
+        )
+
+    object ConsentAgree : FinancialConnectionsInternalEvent(
+        name = "click.agree",
+        mapOf("pane" to Pane.CONSENT.value)
+    )
+
+    class PrepaneClickContinue(
+        pane: Pane
+    ) : FinancialConnectionsInternalEvent(
+        name = "click.prepane.continue",
+        mapOf("pane" to pane.value)
+    )
+
+    class AuthSessionOpened(
+        pane: Pane,
+        flow: String?,
+        defaultBrowser: String?,
+        id: String
+    ) : FinancialConnectionsInternalEvent(
+        "auth_session.opened",
+        mapOf(
+            "auth_session_id" to id,
+            "pane" to pane.value,
+            "flow" to (flow ?: "unknown"),
+            "browser" to (defaultBrowser ?: "unknown")
+        ).filterNotNullValues()
+    )
+
+    override fun toString(): String {
```

Comment:
How come manual toString/equals/hashcode and not a data class?

Reply:
just because it's already a sealed class : (

Reply:
Can we make the sealed classes data classes then?

Reply:
We can't since they don't hold any value - they just transform the received fields into the `FinancialConnectionsAnalyticsEvent` shape (event name + params map).

We can probably remove `toString/equals/hashcode` -  I don't think we're comparing events anywhere across the codebase. I'll take a look!